server/CMakeLists.txt

```diff
@@ -5,7 +5,7 @@ find_package(SFML COMPONENTS system network REQUIRED)
 include_directories(${SFML_INCLUDE_DIR})
 add_executable(ares_server ${source_server})
 
-target_link_libraries(ares_server ${SFML_LIBRARIES})
+target_link_libraries(ares_server ${SFML_LIBRARIES} pthread)
```

Comment:
You should only add pthread on linux to keep the code portable.

I suggest using

```
if (UNIX)
    #do something
endif (UNIX)
```

and

```
if (WIN32)
    #do something
endif (WIN32)
```